src/main/scala/fi/oph/koski/valpas/oppija/ValpasOppijaLaajatTiedotService.scala

```diff
@@ -197,14 +235,16 @@ class ValpasOppijaLaajatTiedotService(
   private def fetchKuntailmoitukset(
     oppija: ValpasOppijaLaajatTiedot
   )(implicit session: ValpasSession): Either[HttpStatus, Seq[ValpasKuntailmoitusLaajatTiedot]] = {
-    timed("fetchKuntailmoitukset", 10) {
-      application.valpasKuntailmoitusService.getKuntailmoitukset(oppija)
-        .map(lisääAktiivisuustiedot(oppija))
+    oppija.mapOppivelvollinen { o =>
```

Comment:
Pohditko muuten  OO-approachia tähän? Yksi vaihtoehto olisi siirtää näitä toimintoja, jotka pitää toimia eri tavalla vapautetulla vs. ei-vapautetulla, ValpasOppijaLaajatTiedot-traitin abstraktien metodien toteutuksiksi. Se saattaisi johtaa koodiin, jossa on pienempi unohduksen vaara, mutta vaikea sanoa, miltä se näyttäisi toteutettuna.